src/Util/BigInteger.php

```diff
@@ -203,4 +231,44 @@ public function compare(BigInteger $y)
     {
         return gmp_cmp($this->value, $y->value);
     }
+
+    /**
+     * Check whether two BigIntegers are the same.
+     *
+     * @param BigInteger $x
+     *
+     * @return bool
+     */
+    public function equals(BigInteger $x)
+    {
+        return 0 === gmp_cmp($this->value, $x->value);
```

Comment:
`gmp_cmp()` is not a constant time function, so you might want to look into something else for doing an equality comparison. - https://paragonie.com/static/audit-reports/2015/12/LCoubucciJWT.pdf

You can check out the code in the `constantTimeCompare()` method here: https://github.com/phpecc/phpecc/pull/114/files